### ISSUE TITLE
lookupElementOfType does not work as intended

### ISSUE THREAD

**Issue by sebinside** [DataFlowAnalysis/DataFlowAnalysis#102]
The method `lookupElementOfType` in the resource provider (e.g., the `PCMURIResourceProvider`) is described to `Looks up an ECore element with the given class type`. However, this only works with root level elements, e.g., Repository, or system model. It does not work recursively with all contained elements (at least in my tests):

```
resourceProvider.lookupElementOfType(RepositoryPackage.eINSTANCE.getRepository()); // This works and returns the repository
resourceProvider.lookupElementOfType(RepositoryPackage.eINSTANCE.getBasicComponent()); // This returns an empty list
``` 

Not only would this feature help when quickly scanning through models, it's also unintuitive as `lookupElementWithId` also works with all non top-level elements.

**TL;DR**: Please make `lookupElementOfType` find *all* elements 🚀

**Comment by Entenwilli** [comment]
This feature is already implemented with the methods `findElement` and `lookupElementWithId` that search though the entire loaded models and does a DFS on their contents. Implementing this Behavior for `lookupElementofType` would work as well, but would lead to performance issues with larger models (especially searching for elements in one of the models loaded last)

**Comment by sebinside** [comment]
Perfect, then it's only a naming / documentation issue and way easier to resolve. Let's discuss this on Tuesday 👍
